pkg/registry/servicecatalog/server/options.go

```diff
@@ -103,6 +105,40 @@ func (o Options) ResourcePrefix() string {
 	return o.EtcdOptions.RESTOptions.ResourcePrefix
 }
 
+// KeyRootFunc returns the appropriate key root function for the storage type in o.
+// This function produces a path that etcd or TPR storage understands, to the root of the resource
+// by combining the namespace in the context with the given prefix
+func (o Options) KeyRootFunc() func(api.Context) string {
+	prefix := o.ResourcePrefix()
+	sType, err := o.StorageType()
+	if err != nil {
+		return nil
+	}
+	if sType == StorageTypeEtcd {
+		return func(ctx api.Context) string {
+			return registry.NamespaceKeyRootFunc(ctx, prefix)
```

Comment:
However, note that:

1.  `NamespaceKeyRootFunc` turns into `NoNamespaceKeyRootFunc` if there is no namespace in the context
2.  The code in `KeyFunc` below is parameterized with whether the resource is namespaced, and chooses the right key function based on that

So, this is not a destructive change.